tests/unit/parsers/test.manifestjson.js

```diff
@@ -2465,25 +2465,59 @@ describe('ManifestJSONParser', () => {
 
   describe('schema error overrides', () => {
     // https://github.com/mozilla/addons-linter/issues/732
-    it('uses a modified error message', () => {
-      const addonLinter = new Linter({ _: ['bar'] });
-      // 'scripts' intentionally misspelled as 'script'.
-      const json = validManifestJSON({
-        background: { script: ['background.js'] },
-      });
-      const manifestJSONParser = new ManifestJSONParser(
-        json,
-        addonLinter.collector
-      );
-      const { errors } = addonLinter.collector;
-      assertHasMatchingError(errors, {
-        code: messages.JSON_INVALID.code,
-        message:
-          '"/background" is not a valid key or has invalid ' +
-          'extra properties',
-      });
-      expect(manifestJSONParser.isValid).toEqual(false);
-    });
+    it.each([
+      [
+        'MV2',
+        {
+          manifest_version: 2,
+          schemaValidatorOptions: {},
+          expectedRequiredProps: '"scripts" or "page"',
+        },
+      ],
+      [
+        'MV3',
+        {
+          manifest_version: 3,
+          schemaValidatorOptions: {
+            maxManifestVersion: 3,
+            enableBackgroundServiceWorker: true,
+          },
+          expectedRequiredProps: '"service_worker", "scripts" or "page"',
+        },
+      ],
+    ])(
+      'reports missing required background properties (%s)',
+      (_variantName, variantTestOptions) => {
+        const {
+          manifest_version,
+          schemaValidatorOptions,
+          expectedRequiredProps,
+        } = variantTestOptions;
+        const addonLinter = new Linter({ _: ['bar'] });
+        // 'scripts' intentionally misspelled as 'script'.
+        const json = validManifestJSON({
+          manifest_version,
+          background: { script: ['background.js'] },
+        });
+        const manifestJSONParser = new ManifestJSONParser(
+          json,
+          addonLinter.collector,
+          { schemaValidatorOptions }
+        );
+        const { errors } = addonLinter.collector;
+        const expectedErrorMessage = `"/background" requires at least one of ${expectedRequiredProps}.`;
+        expect(errors).toEqual(
+          expect.arrayContaining([
+            expect.objectContaining({
+              instancePath: '/background',
+              code: messages.MANIFEST_FIELD_REQUIRED.code,
+              message: expect.stringMatching(expectedErrorMessage),
+            }),
+          ])
+        );
```

Comment:
note: I replaced the previous call to `assertHasMatchingError` with one using toEqual/arrayContaining/objectContaining while I was here, because the latter approach provide a better (more detailed and more likely to be immediately actionable) error message than the [one that hitting an `assertHasMatchingError` logs](https://app.circleci.com/pipelines/github/mozilla/addons-linter/6675/workflows/1cf5015a-4b39-4134-94b5-fc7ef8bbafed/jobs/21014?invite=true#step-106-20192_57).